packages/data/src/world/mq/dodongo_cavern_mq.yml

```diff
@@ -2,159 +2,279 @@
+    #"Dodongo Cavern Lower Lizalfos": "has_explosives_or_hammer || hookshot_anywhere || climb_anywhere"
+  events:
+    MQ_DC_UPPER_LIZALFOS_CLEAR: "soul_lizalfos_dinalfos && (can_use_sticks || has_weapon || can_use_slingshot)"
+  locations:
+    "MQ Dodongo Cavern GS Upper Lizalfos": "gs && (has_explosives_or_hammer || longshot_anywhere)"
+    "MQ Dodongo Cavern Pot Miniboss 1": "true"
+    "MQ Dodongo Cavern Pot Miniboss 2": "true"
+    "MQ Dodongo Cavern Pot Miniboss 3": "true"
+    "MQ Dodongo Cavern Pot Miniboss 4": "true"
+    "MQ Dodongo Cavern Heart Lizalfos Room": "has_explosives_or_hammer || hookshot_anywhere || climb_anywhere" #In Lower section, putting it here too because of the lizalfos event won't trigger when starting in opposite section.
 "Dodongo Cavern Lower Tunnel":
   dungeon: DC
   exits:
-    "Dodongo Cavern Lower Lizalfos": "can_use_bow || ((has_bombflowers || can_use_din) && can_use_slingshot)"
+    "Dodongo Cavern Corridor Before Lower Lizalfos": "event(MQ_DC_LOWER_TUNNEL_EYE_SWITCH)"
+    "Dodongo Cavern Main Room Lower": "true"
+    "Dodongo Cavern Lower Tunnel Side Room": "has_bombflowers || has_explosives_or_hammer || has_blue_fire || has_blue_fire_arrows_mudwall"
+  events:
+    MQ_DC_LOWER_TUNNEL_EYE_SWITCH: "can_use_bow || ((has_bombflowers || can_use_din) && can_use_slingshot)"
   locations:
-    "MQ Dodongo Cavern Tunnel Side Scrub": "business_scrub(0x1e)"
     "MQ Dodongo Cavern Pot East Corridor 1": "true"
     "MQ Dodongo Cavern Pot East Corridor 2": "true"
     "MQ Dodongo Cavern Pot East Corridor 3": "true"
     "MQ Dodongo Cavern Pot East Corridor 4": "true"
+"Dodongo Cavern Lower Tunnel Side Room":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Lower Tunnel": "true"
+  locations:
+    "MQ Dodongo Cavern Tunnel Side Scrub": "business_scrub(0x1e)"
+"Dodongo Cavern Corridor Before Lower Lizalfos":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Lower Lizalfos": "true"
+    "Dodongo Cavern Lower Tunnel": "true"
 "Dodongo Cavern Lower Lizalfos":
   dungeon: DC
   exits:
-    "Dodongo Cavern Lower Tunnel": "soul_lizalfos_dinalfos && (can_use_sticks || has_weapon || can_use_slingshot)"
-    "Dodongo Cavern Poe Room": "soul_lizalfos_dinalfos && (can_use_sticks || has_weapon || can_use_slingshot)"
+    "Dodongo Cavern Corridor Before Lower Lizalfos": "event(MQ_DC_LOWER_LIZALFOS_CLEAR)"
+    "Dodongo Cavern Poe Room": "event(MQ_DC_LOWER_LIZALFOS_CLEAR)"
+    #"Dodongo Cavern Upper Lizalfos": "climb_anywhere || longshot_anywhere"
+  events:
+    MQ_DC_LOWER_LIZALFOS_CLEAR: "soul_lizalfos_dinalfos && (can_use_sticks || has_weapon || can_use_slingshot)"
   locations:
     "MQ Dodongo Cavern Heart Lizalfos Room": "true"
+    "MQ Dodongo Cavern GS Upper Lizalfos": "gs && (climb_anywhere || longshot_anywhere) && (has_explosives_or_hammer || longshot_anywhere)" #In Upper section, putting it here too because of the lizalfos event won't trigger when starting in opposite section.
```

Comment:
Should be able to just climb up the pillar right? in which case you can just have the "can_damage_skull" macro for climb anywhere. No real reason to list longshot anywhere twice either.

"gs && (longshot_anywhere || (climb_anywhere && can_damage_skull))"

Reply:
Changed.